vars/defaultIntegPipeline.groovy

```diff
@@ -28,29 +28,57 @@ def call(Map config = [:]) {
         stages {
             stage('Checkout') {
                 steps {
-                    git branch: "${params.GIT_BRANCH}", url: "${params.GIT_REPO_URL}"
+                    script {
+                        // Allow overwriting this step
+                        if (config.checkoutStep) {
+                            config.checkoutStep()
+                        } else {
+                            git branch: "${params.GIT_BRANCH}", url: "${params.GIT_REPO_URL}"
+                        }
```

Comment:
```suggestion
                    script {
                        config.checkoutStep()
```

I'm not a fan of this pattern, can we have the checkoutStep have a default value be `git branch: "${params.GIT_BRANCH}", url: "${params.GIT_REPO_URL}"`

This changes the model so there is no if/else branch, we always execute the checkout step, but it can be customized.  What do you think of this?

Reply:
> we don't currently use it anywhere

If we don't use it, why add it?  Can we revisit this later if/when its needed?

Reply:
It has been useful for some adhoc testing in the UI to be able to override this directly rather than repeatedly specifying the parameters. I'd like to keep it for that sake, at least until we have a better way to test Jenkins changes locally.